src/store/ChartState.js

```diff
@@ -80,6 +80,11 @@ class ChartState {
             this.stxx.isAutoScale = this.settings && settings.isAutoScale !== false;
         }
         if (this.granularity !== granularity && this.context) {
+            if (calculateTimeUnitInterval(granularity).timeUnit === 'second' && this.mainStore.chartType.isCandle) {
+                chartType = 'mountain';
+            } else {
+                chartType = this.chartType;
```

Comment:
For a reason without this statement when changing time unit from seconds to anything, chart goes back to candle, instead of staying mountain

Reply:
@nazaninreihani it happens because when `chartType` is changed from `candle` to `mountain` in the `if` statement the chart doesn't broadcast the new value of `chartType` to the outer application and the value of `chartType` is still `candle` in the app because of that by changing the granularity from `second` to other ones the application pass the new granularity alongside the previous chartType value. 

You can use the line below after changing `chartType` to 'mountain` at line 84 to broadcast the new type to the outer application.
```
this.chartTypeStore.onChartTypeChanged(chartType);
```

Reply:
I see... great thanks, I'll make the changes